ethereum/networks/src/test/java/tech/pegasys/teku/networks/EphemeryNetworkTest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networks;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.infrastructure.time.SystemTimeProvider;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+import tech.pegasys.teku.spec.config.SpecConfig;
+import tech.pegasys.teku.spec.config.SpecConfigLoader;
+import tech.pegasys.teku.spec.config.SpecConfigReader;
+import tech.pegasys.teku.spec.config.builder.SpecConfigBuilder;
+
+public class EphemeryNetworkTest {
+  private static final long GENESIS_CHAINID = 39438135;
+  private static final long GENESIS_TIMESTAMP = 1720119600;
+  private final SystemTimeProvider timeProvider = new SystemTimeProvider();
+
+  private SpecConfigBuilder builder;
+  private static final long CURRENT_TIMESTAMP = 1725547734;
+  private static final int PERIOD = 28;
+  private static final long PERIOD_IN_SECONDS = (PERIOD * 24 * 60 * 60);
+  private long expectedMinGenesisTime;
+  private long expectedChainId;
+  private final SpecConfigReader reader = new SpecConfigReader();
+
+  @BeforeEach
+  void setUp() {
+    long periodSinceGenesis = EphemeryNetwork.getPeriodsSinceGenesis(timeProvider);
+    expectedMinGenesisTime = GENESIS_TIMESTAMP + (periodSinceGenesis * PERIOD_IN_SECONDS);
+    expectedChainId = GENESIS_CHAINID + periodSinceGenesis;
+    builder = mock(SpecConfigBuilder.class);
+  }
```

Comment:
so yep this is one use... (add final to periods as nit)

The reason we do this though, is because then i can test the first period since genesis, or even pre the first genesis time, and that's the real power of making the timeProvider change, because then we can test with predictableOutput rather than using SystemTimeProvider...

So for example if you wanted 1000 periods since genesis, you could compute a time that gives you that, and then using a StubTimeProvider set that value, and then you're always running at that point in time...

Or before the first genesis...

As a first step having time provider there is adequate, and we can expand these tests separately, i might raise it as a good first issue for someone maybe?

Reply:
Yes, you are right. I have added final and more test cases.